### PR TITLE
[DOCS] Update TOC to match Design Doc

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Changes proposed in this pull request:
- TOC for Docusaurus now matches most recent design doc in superconductive design repo.
- Efforts have been made to create stubs and placeholder documents for documentation that still has not been written.
- Note: Deployment Patterns still has not been fully-fleshed out

### Definition of Done
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: sidebars.js [hunk 1]
```diff
@@ -17,26 +17,61 @@ module.exports = {
...
         'tutorials/getting-started/create-your-first-expectations',
-        'tutorials/getting-started/set-up-data-docs',
+        'tutorials/getting-started/check-out-data-docs',
         'tutorials/getting-started/validate-your-data',
         'tutorials/getting-started/customize-your-deployment'
       ]
     },
+    {
+      type: 'category',
+      label: 'Core Skills',
```
Comment (Aylr) [inline sidebars.js]: The way I read the design spec (which may be incorrect!) is that the core skills are a category within each section, not separate from each section. Can we sync on this?

File: sidebars.js [hunk 2]
```diff
@@ -74,14 +106,27 @@ module.exports = {
...
                 'guides/setup/configuring-data-docs/how-to-host-and-share-data-docs-on-a-filesystem',
                 'guides/setup/configuring-data-docs/how-to-host-and-share-data-docs-on-azure-blob-storage',
                 'guides/setup/configuring-data-docs/how-to-host-and-share-data-docs-on-gcs',
                 'guides/setup/configuring-data-docs/how-to-host-and-share-data-docs-on-amazon-s3'
               ]
-            }
+            },
+            {
+              type: 'category',
+              label: 'Deployment Patterns',
```
Comment (Aylr) [inline sidebars.js]: I don't think deployment patterns belongs in the setup category. The spec seems to indicate that it should be at the same level as "How to Guides".

File: sidebars.js [hunk 3]
```diff
@@ -123,155 +168,130 @@ module.exports = {
...
+                'guides/connecting_to_your_data/advanced/how-to-create-a-batch-from-a-sql-query',
+                'guides/connecting_to_your_data/advanced/how-to-create-a-lightweight-data-catalog-by-applying-a-descriptive-profiler-to-a-configured-datasource',
+                'guides/connecting_to_your_data/advanced/how-to-explore-changes-in-data-over-time-using-a-configured-datasource'
               ]
+            }
+          ]
+        },
+        {
+          type: 'category',
+          label: '🧪 Creating and editing Expectations for your data',
```
Comment (Aylr) [inline sidebars.js:165]: This category seems to be missing all the core skills. Also, the order here does not match the spec: https://github.com/superconductive/design/blob/main/docs/20210421_knoxpod_golden_paths_mental_map/toc.md#creating-custom-expectations-the-article-in-this-section-need-examination-we-might-be-able-to-consolidate

File: sidebars.js [hunk 4]
```diff
@@ -123,155 +168,130 @@ module.exports = {
...
         },
         {
           type: 'category',
           label: '✅ Validating your data',
           items: [
-
             {
               type: 'category',
-              label: 'Working with Validators (name TBD)',
+              label: 'Advanced',
```
Comment (Aylr) [inline sidebars.js:212]: Missing core skills. Order does not match spec: https://github.com/superconductive/design/blob/main/docs/20210421_knoxpod_golden_paths_mental_map/toc.md#icon-validating-your-data

**Review by Aylr — CHANGES_REQUESTED** [review]
Good progress on this tedious and difficult work! We should probably sync before work commences.

File: sidebars.js (hunk 2 above)
Comment (Shinnnyshinshin) [reply sidebars.js]: I need to fix this :)

**Review by Aylr — APPROVED** [review]
Really nice work on this!

**Pull request merged**